Gems/ROS2/Code/Source/SdfAssetBuilder/SdfAssetBuilder.cpp

```diff
@@ -109,18 +114,125 @@ namespace ROS2
         return patterns;
     }
 
+    Utils::UrdfAssetMap SdfAssetBuilder::FindAssets(const urdf::LinkConstSharedPtr& rootLink, const AZStd::string& sourceFilename) const
+    {
+        AZ_Info(SdfAssetBuilderName, "Parsing mesh and collider names");
+        auto assetNames = Utils::GetMeshesFilenames(rootLink, true, true);
+
+        Utils::UrdfAssetMap assetMap;
+
+        using AssetSysReqBus = AzToolsFramework::AssetSystemRequestBus;
+
+        // Unlike the RobotImporter, the SDF Asset Builder does not use the AMENT_PREFIX_PATH
+        // to resolve file locations. There wouldn't be a way to guarantee identical results across
+        // machines or to detect the need to rebuild assets if the environment variable changes.
+        const AZStd::string emptyAmentPrefixPath;
```

Comment:
Why can we not ask the environment for ament path here?
I was not able to perform https://github.com/RobotecAI/o3de-ros2-gem-testing/blob/main/URDFTest/04_IMPORTING_URDF_2/IMPORTING_URDF2_TESTCASE.md test case using URDF builder.
Maybe AMENT_PREFIX_PATH can live in assetinfo?

Reply:
@michalpelka what do you propose to address the test case?

Reply:
I don't actually know what the ament path represents or why it exists, so I'd definitely like to understand the use of ament path further. All I can do is explain my concern with it from an O3DE point of view.

In general, the goal is that O3DE projects should be easily shareable. The project itself should contain both the assets it needs and any configuration needed to process and use those assets. If the asset builder requires environment variables to be set a specific way to locate and process the assets, then the project will presumably be more difficult to share because it will require a manual environment setup outside of O3DE. This will also make things like configuring build machines that much more difficult, because they would require extra manual configuration steps for each project.

That said, we _already_ have a dependency on the currently-active ROS version (via `source`), so if this is a similar type of dependency where everyone is expected to have their environments configured the same way, maybe that's OK.

However, the design that we've currently got proposed is that we'll have settings files globally for the asset builder as well as optionally per-file. The first setting that we'll add is a way to directly map URI prefixes to relative directories. This would replace the current use of environment variables and search heuristics. The benefits are that it allows us to solve some of the cases that the current heuristics seemingly fail to find and it makes the settings themselves shareable through the project files. The main drawback is that it requires manual setup.

The settings file would look something like this:
```
{
    "uri_prefixes": {
        "model://": "Assets/my_robot/models/",
        "package://": "Assets/",
        "http://": "Assets/web/",
        "aws/models":  "Assets/aws/models/"
    }
}
```

Thoughts?

(Also, thank you to the pointer to the test cases, I wasn't aware these existed. FWIW, here's the 'in the wild' model that I was using to test this: https://github.com/ANYbotics/anymal_b_simple_description ).